routing/router.go

```diff
@@ -1803,16 +1803,94 @@ type routingMsg struct {
 	err chan error
 }
 
+// RouteRequest contains the parameters for a pathfinding request.
+type RouteRequest struct {
+	// Source is the node that the path originates from.
+	Source route.Vertex
+
+	// Target is the node that the path terminates at.
+	Target *route.Vertex
+
+	// Amount is the amount in millisatoshis to be delivered to the target
+	// node.
+	Amount lnwire.MilliSatoshi
+
+	// TimePreference expresses the caller's time preference for
+	// pathfinding.
+	TimePreference float64
+
+	// Restrictions provides a set of additional restrictions that the
+	// route must adhere to.
+	Restrictions *RestrictParams
+
+	// CustomRecords is a set of custom tlv records to include for the
+	// final hop.
+	CustomRecords record.CustomSet
+
+	// RouteHints contains an additional set of edges to include in our
+	// view of the graph.
+	RouteHints RouteHints
+
+	// FinalExpiry is the cltv delta for the final hop.
+	FinalExpiry uint16
+}
+
+// RouteHints is an alias type for a set of route hints, with the source node
+// as the map's key and the details of the hint(s) in the edge policy.
+type RouteHints map[route.Vertex][]*channeldb.CachedEdgePolicy
+
+// NewRouteRequest produces a new route request.
+func NewRouteRequest(source route.Vertex, target *route.Vertex,
+	amount lnwire.MilliSatoshi, timePref float64,
+	restrictions *RestrictParams, customRecords record.CustomSet,
+	routeHints RouteHints, finalExpiry uint16) *RouteRequest {
+
+	return &RouteRequest{
+		Source:         source,
+		Target:         target,
+		Amount:         amount,
+		TimePreference: timePref,
+		Restrictions:   restrictions,
+		CustomRecords:  customRecords,
+		RouteHints:     routeHints,
+		FinalExpiry:    finalExpiry,
+	}
+}
+
+// validate ensures that a route request is logically sane.
+func (r *RouteRequest) validate() error {
+	if r.Target == nil {
+		return errors.New("target node required")
+	}
+
+	return nil
+}
+
+func (r *RouteRequest) target() route.Vertex {
+	return *r.Target
+}
+
+func (r *RouteRequest) hints() RouteHints {
```

Comment:
These methods are introduced because we're using a single request to represent regular and blinded payments. 

Take target as an example: 
- For a regular payment we provide a pubkey.
- For a blinded payment we extract the final pubkey from the blinded route. 

The method is a way for callers to interact with the multi-purpose request without needing to worry about those details. Things like `Amount` are different, because we use use that value as-is. 

I made them private because they're not needed outside of `router`, but I agree it's kind of confusing. I also thought about interfacing this out, but it ended up much more boiler-plate-ey. 

For now, going to update this with clearer comments and unexport the fields that we use in the private methods. 
Open to suggestions on how to improve this!